tiledb/tests/test_libtiledb.py

```diff
@@ -3923,6 +3923,32 @@ def test_ctx_thread_cleanup(self):
             if n == 0:
                 start_threads = len(thisproc.threads())
 
+    def test_delete_fragments(self):
+        dshape = (1, 3)
+        num_writes = 10
+
+        def create_array(target_path, dshape):
+            dom = tiledb.Domain(tiledb.Dim(domain=dshape, tile=len(dshape)))
+            att = tiledb.Attr(dtype="int64")
+            schema = tiledb.ArraySchema(domain=dom, attrs=(att,), sparse=True)
+            tiledb.libtiledb.Array.create(target_path, schema)
+
+        def write_fragments(target_path, dshape, num_writes):
+            for i in range(1, num_writes + 1):
+                with tiledb.open(target_path, "w", timestamp=i) as A:
+                    A[[1, 2, 3]] = np.random.rand(dshape[1])
```

Comment:
Would suggest adding a functional test here as well. Write a single coordinate at a time based on `i`, and then do a read at the end and make sure the deleted coordinates are gone. eg `A[[i]] = ...` in the loop.

Then, before delete we would expect coordinates 1..10, but after delete we should get 1,2 and 7,8,9,10.